src/Data/HasOrigin.php

```diff
@@ -58,18 +59,18 @@ public function value($key)
 
     public function origin($origin = null)
     {
-        return $this->fluentlyGetOrSet('origin')
-            ->getter(function ($origin) {
-                return $origin
-                    ? Blink::once($this->getOriginBlinkKey(), fn () => $this->getOriginByString($origin))
-                    : null;
-            })
-            ->setter(function ($origin) {
-                Blink::forget($this->getOriginBlinkKey());
-
-                return is_object($origin) ? $this->getOriginIdFromObject($origin) : $origin;
-            })
-            ->args(func_get_args());
+        if (func_num_args() === 0) {
+            return $this->origin
+                ? Blink::once($this->getOriginBlinkKey(), fn () => $this->getOriginByString($this->origin))
+                : null;
+        }
+
+        Blink::forget($this->getOriginBlinkKey());
+
+        $this->origin = is_object($origin) ? $this->getOriginIdFromObject($origin) : $origin;
+        $this->cachedHasOrigin = $this->origin != null;
```

Comment:
Maybe this line can be moved into `hasOrigin()` and then you can just literally call `$this->hasOrigin()` here.

Reply:
I took another look at that cached call logic with a less-tired brain and just refactored it to `$this->origin != null`. Let me know if you think this is not going to work 🤞